src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,2532 @@
+    [InlineData("", "[System.Runtime.CompilerServices.OverloadResolutionPriority(1)]")]
+    public void PartialMethod(string definitionPriority, string implementationPriority)
+    {
+        var definition = $$"""
+            public partial class C
+            {
+                {{definitionPriority}}
+                public partial void M(object o);
+            }
+            """;
+
+        var implementation = $$"""
+            public partial class C
+            {
+                {{implementationPriority}}
+                public partial void M(object o) => System.Console.Write(1);
+                public void M(string s) => throw null;
+            }
+            """;
+
+        var executable = """
+            var c = new C();
+            c.M("");
+            """;
+
+        CompileAndVerify([executable, definition, implementation, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "1").VerifyDiagnostics();
+    }
+
+    [Theory]
+    [InlineData("[System.Runtime.CompilerServices.OverloadResolutionPriority(1)]", "")]
+    [InlineData("", "[System.Runtime.CompilerServices.OverloadResolutionPriority(1)]")]
+    public void PartialIndexer(string definitionPriority, string implementationPriority)
+    {
+        var definition = $$"""
+            public partial class C
+            {
+                {{definitionPriority}}
+                public partial int this[object o] { get; set; }
+            }
+            """;
+
+        var implementation = $$"""
+            public partial class C
+            {
+                {{implementationPriority}}
+                public partial int this[object o]
+                {
+                    get { System.Console.Write(1); return 1; }
+                    set => System.Console.Write(2);
+                }
+                public int this[string s]
+                {
+                    get => throw null;
+                    set => throw null;
+                }
+            }
+            """;
+
+        var executable = """
+            var c = new C();
+            _ = c[""];
+            c[""] = 0;
+            """;
+
+        CompileAndVerify([executable, definition, implementation, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "12").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void AttributeAppliedTwiceMethod_Source()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            var c = new C();
+            c.M("");
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                [OverloadResolutionPriority(2)]
+                public void M(object o) => System.Console.Write(1);
+                public void M(string s) => System.Console.Write(2);
+            }
+            """;
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
```

Comment:
You probably need to use a custom definition of the attribute with AllowMultiple=true. Otherwise, we hit this early check

Reply:
What was the resolution?